protocols/kad/src/behaviour.rs

```diff
@@ -283,6 +315,18 @@ impl<TSubstream> Kademlia<TSubstream> {
         self.start_query(QueryInfoInner::GetProviders { target, pending_results: Vec::new() });
     }
 
+    /// Starts an iterative `GET_VALUE` request.
+    pub fn get_data(&mut self, key: Multihash) {
+        self.start_query(QueryInfoInner::GetValue { key, results: vec![] });
+    }
+
+    /// Starts an iterative `PUT_VALUE` request
+    pub fn put_data(&mut self, key: Multihash, data: &[u8]) {
```

Comment:
```suggestion
    pub fn put_value(&mut self, key: &Multihash, value: Vec<u8>) {
```

Reply:
How about taking `data` as a `Vec<u8>`, i.e. let the function take ownership? I think that may avoid a clone if the caller of the function no longer needs the value afterwards, and taking ownership of the values seems to be common for "insert" operations on data structures. Similarly, taking the key by reference gives a bit more leeway in the implementation itself, in case it may not always need the key by-value.

Reply:
I will bring back taking `key` by value since that's what the other methods do